File: sharp_brain/sharp_brain/Neural/Neuron.cs
namespace SharpBrain.Neural
{
    public abstract class Neuron
    {
        protected static readonly double ETA = 1.0;

        protected double netInput = 0.0;
        protected double delta = 0.0;
        protected int layerIndex = 0;
        protected int index = 0;

        public double Value { get; set; }

        public double NetInput
        {
            get { return netInput; }
        }

        public double Delta
        {
            get { return delta; }
        }

        protected Neuron(int layerIndex, int index)
        {
            this.layerIndex = layerIndex;
            this.index = index;
        }

        public abstract double Activate();
    }
}

File: java_brain/src/main/java/org/netbrains/javabrain/neural/Neuron.java
package org.netbrains.javabrain.neural;

import java.util.List;

public abstract class Neuron {
    protected static final double eta = 0.5;

    protected int layerIndex;
    protected int index;
    protected double value;
    protected double netInput;
    protected double delta;
    protected List<Connection> connections;

    public Neuron(int layerIndex, int index) {
        this.layerIndex = layerIndex;
        this.index = index;
        value = 0.0;
        delta = 0.0;
    }

    public double getValue() {
        return value;
    }

    public double getDelta() {
        return delta;
    }

    public List<Connection> getConnections() {
        return connections;
    }

    public Connection get(int index) {
        return connections.get(index);
    }

    public void setConnections(List<Connection> connections) {
        this.connections = connections;
    }

    public abstract double activate();
}
